File: Backend/src/main/java/com/cinema/ticketsystem/dto/ScreeningDetailDTO.java
package com.cinema.ticketsystem.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ScreeningDetailDTO {
    private Long id;
    private Long cinemaId;
    private String cinemaName;
    private Long movieId;
    private String movieTitle;
    private LocalDateTime startDateTime;
    private BigDecimal ticketPrice;
    private Integer rows;
    private Integer seatsPerRow;
    private List<SeatDTO> seats;
}

File: cinema-ticket-C#/Dto/Screening/ScreeningDetailDto.cs
namespace CinemaTicket.Dto.Screening;

public record ScreeningDetailDto(
    long Id,
    long CinemaId,
    string CinemaName,
    int CinemaRows,
    int CinemaSeatsPerRow,
    long MovieId,
    string MovieTitle,
    string MovieDescription,
    int MovieDurationMinutes,
    DateTime StartDateTime,
    decimal TicketPrice,
    int AvailableSeats,
    int TotalSeats
);
